include/openthread/crypto.h

```diff
@@ -74,6 +74,18 @@ struct otCryptoSha256Hash
  */
 typedef struct otCryptoSha256Hash otCryptoSha256Hash;
 
+#if OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE
```

Comment:
I noticed that that this PR is adding some checks for `#if OPENTHREAD_CONFIG_{NAME}` in some of the OT public header and/or platform header (e.g., `radio.h`).

This is generally avoided with OT, i.e. we don't include such `#if` checks in any of public or platform headers. Note that none of the public or platform header file include the `"openthread-core-config.h"` (see below for more background/reason for this). 

Here is the common aproach we use in public/platform headers:
- In public OT API, we document if an entire set of APIs or a specific OT API requires certain config option. There are many examples of this. 
https://github.com/openthread/openthread/blob/f4b188d79dafdbde081497fcd116fa843fb2be37/include/openthread/dataset_updater.h#L49-L52

- In the case of types (`struct`, `enum`, callback, etc) definitions in public/platform header, the types are generally included without any `#if` check and it is documented if/how their use is tied to some config option
https://github.com/openthread/openthread/blob/f4b188d79dafdbde081497fcd116fa843fb2be37/include/openthread/link.h#L61-L75

I have not had the chance to look in detail in this PR and PSA implementation, but wanted to point this out so we can consider how to address this.

-----

This older issue https://github.com/openthread/openthread/issues/4333 points to some background on why we aim to avoid use of `#if` config check in the public/platform headers. It mainly boils down to how OT can be integrated in projects and products. 
- OT can be built as a library which is then linked with other modules (application layer, platform implementation, OS, etc.) to form the device's firmware or an executable.
   - The OT public (and platform) header file are exposed to other modules and can be included in other modules so that they can use OT APIs. 
- The `OPENTRHEAD_CONFIG` are intended for project integrator to configure OT library itself. 
  - While some of the bigger features can be configured (enabled) through cmake or autoconf build config options, there are many configs (paramters of smaller feature) that are defined in a config header file.
- The project integrator can provide a project specific OT config header file (through the OPENTHREAD_PROJECT_CORE_CONFIG_FILE or cmake `OT_CONIFG`) when build OT library.
- In such a case, such a project-specific header file is not owned by OT code and therefore we cannot expose it to other libraries as a public header file.

Reply:
Thank you for the comment @abtink I did notice this, and was wondering how to do this. I will have a look and make the necessary changes.